src/Microsoft.AspNetCore.Sockets.Http/Internal/Transports/WebSocketsTransport.cs

```diff
@@ -18,9 +19,9 @@ public class WebSocketsTransport : IHttpTransport
         private readonly WebSocketOptions _options;
         private readonly ILogger _logger;
         private readonly Channel<byte[]> _application;
-        private readonly string _connectionId;
+        private readonly ConnectionContext _connection;
```

Comment:
Change to `DefaultConnectionContext` and just access the property. Or pass the mode in directly.